charts/cluster-api-cluster-openstack/templates/cluster.yaml

```diff
@@ -37,10 +37,9 @@ metadata:
     # Let CAPO do this in its chosen order.
     argocd.argoproj.io/sync-options: Delete=false
 spec:
-  cloudName: {{ .Values.openstack.cloud }}
   identityRef:
     name: {{ include "cloudconfig.name" . }}
-    kind: Secret
+    cloudName: {{ .Values.openstack.cloud }}
```

Comment:
Okay, now this is an intriguing change... references in Kubernetes are supposed to be well known defined types e.g. https://github.com/kubernetes/api/blob/master/core/v1/types.go#L6578